samza-core/src/main/scala/org/apache/samza/system/chooser/BootstrappingChooser.scala

```diff
@@ -120,15 +136,24 @@ class BootstrappingChooser(
   override def register(systemStreamPartition: SystemStreamPartition, offset: String) {
     debug("Registering stream partition with offset: %s, %s" format (systemStreamPartition, offset))
 
-    // If the offset we're starting to consume from is the same as the upcoming 
-    // offset for this system stream partition, then we've already read all
-    // messages in the stream, and we're at head for this system stream 
-    // partition.
-    checkOffset(systemStreamPartition, offset, OffsetType.UPCOMING)
-
     wrapped.register(systemStreamPartition, offset)
 
-    registeredSystemStreamPartitions += systemStreamPartition
+    val system = systemStreamPartition.getSystem
+    val systemAdmin = systemAdmins.getOrElse(system, throw new SamzaException("SystemAdmin is undefined for System: %s" format system))
+    /**
+     * SAMZA-1100: When a input SystemStream is consumed as both bootstrap and broadcast
+     * BootstrappingChooser should record the lowest offset for each registered SystemStreamPartition.
+     * When multiple tasks processing a SystemStreamPartition runs within a container
+     * and share the same chooser, then the lowest offset should be chosen as starting offset.
+     */
+    if (!registeredSystemStreamPartitions.contains(systemStreamPartition)) {
+      registeredSystemStreamPartitions += systemStreamPartition -> offset
+    } else if (offset != null) {
+      val comparatorResult: Integer = systemAdmin.offsetComparator(registeredSystemStreamPartitions(systemStreamPartition), offset)
```

Comment:
SystemAdmin.offsetComparator returns null(as mentioned in javadoc of the interface), when the offsets are incomparable. In that case, this patch already chooses the offset provided by the first task for a ssp(lineNo: 153). 

Adding a log message at warn level. I don't want the chooser to fail & throw up the exception.